### ISSUE TITLE
Cant dynmicly change the weights of a multi-output model loss functions

### ISSUE THREAD

**Issue by Ram-WD** [tensorflow/tensorflow#52230]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):2.4
- Are you willing to contribute it (Yes/No):yes

**Describe the feature and the current behavior/state.**
I have a multi output model using the sub classing API where each output have a different loss function, and i want to change the loss functions weights dynamically in the training  process ( as shown : https://arxiv.org/pdf/1711.02257.pdf  and more papers). 
It is reasonable to have the option to alter these weights using a callback
 ( very much like the learning rate scheduler https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/LearningRateScheduler)  

here is the call() method that my sub-class model uses: 
NOTE : the ges_out is a classifier output 
```
    def call(self, inputs, training=False, **kwargs):

        # intro block1
        x = self.conv1(inputs)
        x = self.spatial_dropout1(x, training=training)
        x = self.conv2(x)
        x = self.spatial_dropout2(x, training=training)
        # x = self.avgpool1(x)
        x = self.conv3(x)
        intro_out = self.spatial_dropout3(x, training=training)
        # intro_out = self.avgpool2(x)

        # classification block

        g = self.conv_ges1(intro_out)
        g = self.spatial_dropout_ges1(g)
        g = self.maxpool_ges1(g)
        g = self.conv_ges2(g)
        g = self.spatial_dropout_ges2(g)
        g = self.maxpool_ges2(g)
        g = self.flatten_ges(g)
        g = self.fc_ges1(g)
        g = self.dropout_ges1(g, training=training)
        g = self.fc_ges2(g)
        g = self.dropout_ges2(g, training=training)
        ges_out = self.fc_ges_out(g)

        # segmentation block
        s = self.conv_seg1(intro_out)
        s = self.spatial_dropout_seg1(s)
        s = self.avgpool_seg1(s)
        s = self.conv_seg2(s)
        s = self.spatial_dropout_seg2(s)
        s = self.avgpool_seg2(s)
        s = self.flatten_seg(s)
        s = self.fc_seg1(s)
        s = self.dropout_seg1(s, training=training)
        seg_out = self.fc_seg_out(s)

        return {"gesture_out": ges_out, "segmentation_out": seg_out}
```

and here is how i use model_compile(...):
  ```
  def model_compile_and_summary():
        metrics = {"gesture_out": ["accuracy"], "segmentation_out": ["mean_squared_error"]}
        loss_functions = {"gesture_out": keras.losses.CategoricalCrossentropy(from_logits=True),
                              "segmentation_out": keras.losses.MeanAbsoluteError()}
        loss_weights = {"gesture_out": 1.0, "segmentation_out": 0.0001}
        optimizer_name = "Adam"
        optimizer = get_optimizer_from_name(optimizer_name)
        MyModel.model.compile(optimizer=optimizer, loss=loss_functions, metrics=metrics, loss_weights=loss_weights)
```
  
**Will this change the current API? How?**
the current API have the option to pass constant loss functions weights in the model.compile(...) method.  
i assigned each output to each loss using a dictionary   
**Who will benefit with this feature?**
 i think all the researchers/engineers  that currently deal with multi-task learning.

**Comment by mohantym** [comment]
Hi @Ram-WD! 
Please post this feature request on [keras-team/keras repo](https://github.com/keras-team/keras/issues) too .
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!
